core/src/main/scala/com/banno/kafka/RecordStream.scala

```diff
@@ -595,6 +723,66 @@ object RecordStream {
         }
     }
 
+    private[RecordStream] type NeedsConsumer[F[_], A] =
+      Function[ConsumerApi[F, GenericRecord, GenericRecord], A]
+
+    private[RecordStream] case class StreamSelectorImpl[F[_]: Async, P[_[_], _], A](
+        whetherCommits: WhetherCommits[P],
+        topical: Topical[A, ?],
+    ) extends StreamSelector[F, NeedsConsumer[F, *], P, IncomingRecords[A]] {
+      override val present: NeedsConsumer[F, P[F, IncomingRecords[A]]] =
+        c => whetherCommits.extrude(recordStream(c, topical))
+
+      override val pastAndPresent: NeedsConsumer[F, PastAndPresent.Batched[F, P, A]] =
+        consumer =>
+          whetherCommits.pastAndPresent(
+            history = history(consumer),
+            present = present(consumer),
+          )
+
+      override val history: NeedsConsumer[F, Stream[F, IncomingRecords[A]]] =
+        _.recordsThroughAssignmentLastOffsetsOrZeros(
+              pollTimeout,
+              10,
+              commitMarkerAdjustment = true
+            )
+            .prefetch
+            .evalMap(parseBatch(topical))
+
+    }
+
+    private def assign[F[_]: Monad, A, B](
+      consumer: ConsumerApi[F, GenericRecord, GenericRecord],
+      topical: Topical[A, B],
+      seekToF: Kleisli[F, PartitionQueries[F], SeekTo],
+    ): F[Unit] =
+      for {
+        seekTo <- seekToF(consumer.partitionQueries)
+        () <- consumer.assignAndSeek(topical.names.map(_.show).toList, seekTo)
+        // By definition, no need to ever read old ephemera.
+        () <- topical.aschematic.traverse_(ephemeralTopicsSeekToEnd(consumer))
+      } yield ()
+
+    private[RecordStream] def assignAndSeek[F[_]: Async, P[_[_], _], X](
+        baseConfigs: BaseConfigs[P],
+        topical: Topical[X, ?],
+    ): SelectAndSeek[F, P, IncomingRecords[X]] =
+      StreamSelectorImpl(baseConfigs.whetherCommits, topical).mapK(
+        new ~>[NeedsConsumer[F, *], SeekResource[F, *]] {
+          override def apply[A](fa: NeedsConsumer[F, A]): SeekResource[F, A] =
+            new Seeker[F, Resource[F, A]] {
+              override implicit val F: Applicative[F] = Applicative[F]
```

Comment:
I think it would be better to have an `Impl` for Seeker. Then might have a shot at `FunctionK.lift` and cleaning up this egregious syntax.